aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/DeleteDirStepTest.java

```diff
@@ -0,0 +1,128 @@
+        File f = new File(workspace);
+        Assert.assertFalse("Workspace driectory should no longer exist", f.exists());
+    }
+
+    @Test
+    public void testDeleteTopLevelDir() throws Exception {
+        String workspace = runAndGetWorkspaceDir(
+                "node {\n" +
+                "  writeFile file: 'f1', text: 'some text'\n" +
+                "  writeFile file: 'f2', text: 'some text'\n" +
+                "  writeFile file: '.hidden', text: 'some text'\n" +
+                "  writeFile file: 'sub1/f1', text: 'some text'\n" +
+                "  writeFile file: '.sub2/f2', text: 'some text'\n" +
+                "  writeFile file: '.sub3/.hidden', text: 'some text'\n" +
+                "  echo 'workspace is ---' + pwd() + '---'\n" + 
+                "  deleteDir()\n" + 
+                "}");
+        File f = new File(workspace);
+        Assert.assertFalse("Workspace driectory should no longer exist", f.exists());
+    }
+
+    @Test
+    public void testDeleteSubFolder() throws Exception {
+        String workspace = runAndGetWorkspaceDir(
+                "node {\n" +
+                "  writeFile file: 'f1', text: 'some text'\n" +
+                "  writeFile file: 'f2', text: 'some text'\n" +
+                "  writeFile file: '.hidden', text: 'some text'\n" +
+                "  writeFile file: 'sub1/f1', text: 'some text'\n" +
+                "  writeFile file: '.sub2/f2', text: 'some text'\n" +
+                "  writeFile file: '.sub3/.hidden', text: 'some text'\n" +
+                "  echo 'workspace is ---' + pwd() + '---'\n" + 
+                "  dir ('sub1') {\n" +
+                "    deleteDir()\n" +
+                "  }" + 
+                "}");
+
+        File f = new File(workspace);
+        Assert.assertTrue("Workspace directory should still exist", f.exists());
+        Assert.assertTrue("f1 should still exist", new File(f, "f1").exists());
+        Assert.assertTrue("f1 should still exist", new File(f, "f2").exists());
+        Assert.assertFalse("sub1 should not exist", new File(f, "sub1").exists());
+        Assert.assertTrue(".sub2/f2 should still exist", new File(f, ".sub2/f2").exists());
+        Assert.assertTrue(".sub3/.hidden should still exist", new File(f, ".sub3/.hidden").exists());
+    }
+
+
+    /**
+     * Runs the given flow and returns the workspace used.
+     * @param flow a flow definition.
+     * @return the workspace used.
+     */
+    private String runAndGetWorkspaceDir(String flow) throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+
+        p.setDefinition(new CpsFlowDefinition(flow));
+        QueueTaskFuture<WorkflowRun> future = p.scheduleBuild2(0);
+        r.assertBuildStatusSuccess(future);
+        WorkflowRun run = future.get();
+        
+        String workspace = null;
+        // what would be really nice is the equivalent of a head rather than a tail...
+        List<String> logs = run.getLog(50);
```

Comment:
`JenkinsRule.getLog` is the nondeprecated way to get the complete log from a test.

Anyway it is much easier to call:

```java
FilePath ws = r.jenkins.getWorkspaceFor(p);
```